src/services/rag/components/retrievers/dense.py

```
@@ -125,20 +132,28 @@ async def process(self, query: str, kb_name: str, **kwargs) -> Dict[str, Any]:
             if not embeddings_file.exists():
                 self.logger.error(f"Embeddings file not found: {embeddings_file}")
                 return self._empty_response(query)
-            
+
             with open(embeddings_file, "rb") as f:
                 embeddings = pickle.load(f)
```

Ruff (0.14.10)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)